back-end/src/main/kotlin/Gneiss/PacketCompiler/DatabaseAccess/UserDao.kt

```diff
@@ -35,25 +35,26 @@ class UserDao {
 
     fun validateCredentials(email: String, password: String): CredentialsResponse {
         getConnection()
-        var prepStatement = connection!!.prepareStatement(validateCredentialsQuery)
+        var prepStatement = connection!!.prepareStatement(validateCredentialsQuery, ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_UPDATABLE)
         prepStatement.setString(1, email)
         prepStatement.setString(2, password)
         var resultSet = prepStatement.executeQuery()
 
         // Check how many rows are in the result set, if it equal to one it is a valid set of credentials
         var rowCount = getRowCount(resultSet)
-        connection!!.close()
 
         // If the credentials are valid, we need to get the role_id out of the result
         // To do this, we reset the result set pointer to the first row with beforeFirst() and next()
         // And then get the int in the fourth position, where role_id is the query
         if (rowCount == 1) {
             resultSet.beforeFirst()
             resultSet.next()
-            val roleId = resultSet.getInt(4)
-            return CredentialsResponse(true, roleId.toString())
+            val roleId = resultSet.getString(4)
+            connection!!.close()
```

Comment:
Normally for closing resources, a finally block, or Kotlin's [use](https://www.baeldung.com/kotlin/try-with-resources) function is best, but I don't see much of a risk of anything here throwing, so I think this is fine.

Another benefit of using one of the strategies above is knowing at compile time if we're going to be using a resource that's been closed.